metloom/pointdata/snotel.py

```diff
@@ -82,7 +82,9 @@ def _snotel_response_to_df(self, result_map: Dict[SensorDescription, List[dict]]
             # set index so joining works
             sensor_df.set_index("datetime", inplace=True)
             sensor_df = sensor_df.filter(final_columns)
-            df = join_df(df, sensor_df, filter_unused=True)
+            # filter to rows that have value
+            sensor_df = sensor_df.loc[pd.notna(sensor_df[variable.name])]
```

Comment:
I also added this before the merge. We shouldn't be returning empty values for a sensor. The index, column combo will be set to nan if another sensor has data for this index anyway. This just avoids returning dates with no data at all